compute/resource_pipeline.go

```diff
@@ -0,0 +1,231 @@
+			if err != nil {
+				return resource.NonRetryableError(err)
+			}
+			state := *i.State
+			if state == desiredState {
+				return nil
+			}
+			return resource.RetryableError(fmt.Errorf("Pipeline %s is in state %s, not yet in state %s", id, state, desiredState))
+		})
+}
+
+func (a pipelinesAPI) waitForDeleted(id string, timeout time.Duration) error {
+	return resource.RetryContext(a.ctx, timeout,
+		func() *resource.RetryError {
+			i, err := a.read(id)
+			if err != nil {
+				if e, ok := err.(common.APIError); ok && e.IsMissing() {
+					return nil
+				}
+				return resource.NonRetryableError(err)
+			}
+			return resource.RetryableError(fmt.Errorf("Pipeline %s is in state %s, not yet deleted", id, *i.State))
+		})
+}
+
+func adjustPipelineResourceSchema(m map[string]*schema.Schema) map[string]*schema.Schema {
+	clusters, _ := m["clusters"].Elem.(*schema.Resource)
+	clustersSchema := clusters.Schema
+	clustersSchema["spark_conf"].DiffSuppressFunc = sparkConfDiffSuppressFunc
+
+	awsAttributes, _ := clustersSchema["aws_attributes"].Elem.(*schema.Resource)
+	awsAttributesSchema := awsAttributes.Schema
+	delete(awsAttributesSchema, "first_on_demand")
+	delete(awsAttributesSchema, "availability")
+	delete(awsAttributesSchema, "spot_bid_price_percent")
+	delete(awsAttributesSchema, "ebs_volume_type")
+	delete(awsAttributesSchema, "ebs_volume_count")
+	delete(awsAttributesSchema, "ebs_volume_size")
+
+	m["libraries"].MinItems = 1
+
+	return m
+}
+
+// ResourcePipeline ...
+func ResourcePipeline() *schema.Resource {
+	var pipelineSchema = common.StructToSchema(pipelineSpec{}, adjustPipelineResourceSchema)
+	return common.Resource{
+		Schema: pipelineSchema,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var s pipelineSpec
+			err := common.DataToStructPointer(d, pipelineSchema, &s)
+			if err != nil {
+				return err
+			}
+			api := newPipelinesAPI(ctx, c)
+			id, err := api.create(s, false)
+			if err != nil {
+				return err
+			}
+			if err = api.waitForState(id, d.Timeout(schema.TimeoutCreate), StateRunning); err != nil {
+				return err
+			}
+			d.SetId(id)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			i, err := newPipelinesAPI(ctx, c).read(d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(*i.Spec, pipelineSchema, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var s pipelineSpec
+			if err := common.DataToStructPointer(d, pipelineSchema, &s); err != nil {
+				return err
+			}
+			return newPipelinesAPI(ctx, c).update(d.Id(), s, false)
+		},
+		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			api := newPipelinesAPI(ctx, c)
+			if err := api.delete(d.Id()); err != nil {
+				return err
+			}
+			return api.waitForDeleted(d.Id(), d.Timeout(schema.TimeoutDelete))
```

Comment:
on the second thought - can you put `waitForDeleted` code into `delete` method? this way we always have api-wrapper-level consistency for deleting the entity. which will make a good GoSDK in the future if we decide to